code/modules/client/preference_setup/general/02_body.dm

```diff
@@ -540,29 +566,21 @@ var/global/list/valid_bloodtypes = list("A+", "A-", "B+", "B-", "AB+", "AB-", "O
 					pref.rlimb_data[second_limb] = null
 
 			if("Prosthesis")
-				var/decl/species/temp_species = get_species_by_key(pref.species || GLOB.using_map.default_species)
-				var/tmp_bodytype = temp_species.get_bodytype(user)
-				var/list/usable_manufacturers = list()
-				for(var/company in chargen_robolimbs)
-					var/datum/robolimb/M = chargen_robolimbs[company]
-					if(tmp_bodytype in M.bodytypes_cannot_use)
-						continue
-					if(length(M.species_restricted) && !(temp_species.name in M.species_restricted))
-						continue
-					if(M.applies_to_part.len && !(limb in M.applies_to_part))
-						continue
-					if(M.allowed_bodytypes && !(tmp_bodytype in M.allowed_bodytypes))
-						continue
-					usable_manufacturers[company] = M
-				if(!usable_manufacturers.len)
+				var/list/usable_manufacturers
+				var/list/all_robolimbs = decls_repository.get_decls_of_type(/decl/prosthetics_manufacturer)
+				for(var/limb_type in all_robolimbs)
+					var/decl/prosthetics_manufacturer/R = new limb_type()
```

Comment:
that doesn't look very kosher, why instancizing decls all over again every time here?

Reply:
Not sure how that snuck in, oops.